contracts/uniswap/AnteUniswapGetReservesTest.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../AnteTest.sol";
+import "../interfaces/IERC20.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol";
+
+
+/// @title Ante Uniswap Get Reserves
+/// @notice Ensures that getReserves returns a value that makes sense 
+contract AnteUniswapGetReservesTest is AnteTest("Ensure that getReserves returns a reasonable value") {
+
+    AggregatorV3Interface private immutable priceFeed;
+    IUniswapV2Pair private uniswapPair;
+
+    uint256 private immutable decimals0;
+    uint256 private immutable decimals1;
+    address private immutable chainLinkOracle;
+
+    uint256 private lastCheckBlock = 0;
+    int256 private lastCheckPercentage = 0;
+
+    /// @param _uniswapPair The Uniswap pair to test.
+    /// @param _chainLinkOracle price feed to use.
+    /// @notice The Chainlink order should be expensive currency -> cheaper currency. 
+    /// Eg ETH/USD and not the other way around
+    constructor(address _uniswapPair, address _chainLinkOracle ) {
+        protocolName = "Uniswap";
+
+        testedContracts = [_uniswapPair];
+        chainLinkOracle = _chainLinkOracle;
+
+        priceFeed = AggregatorV3Interface(_chainLinkOracle);
+        uniswapPair = IUniswapV2Pair(_uniswapPair);
+
+        decimals0 = IERC20(uniswapPair.token0()).decimals();
+        decimals1 = IERC20(uniswapPair.token1()).decimals();
+    }
+
+    /// @notice Pre-calls the function as a flash loan attack prevention
+    function preCall() public {
+        (uint112 reserve0, uint112 reserve1,) = uniswapPair.getReserves();
+        (, int256 price, , ,) = priceFeed.latestRoundData();
+
+        lastCheckBlock = block.number;
+        lastCheckPercentage = calculatePercentage(reserve0, reserve1, price, decimals0, decimals1);
+    }
+
+    /// @notice Test must be called 2 to 50 block after the preCall
```

Comment:
Is there a reason why we there's a 50 block cutoff after the precall?

Reply:
Oh right this was the upper bounds that you suggested be removed. I'll remove this.

Reply:
Update the notice to our condition please